### PR TITLE
APEXCORE-519 Added DIGEST authentication

### PR DESCRIPTION
**Pull request by pramodin**
@vrozov please see

### CODE REVIEW

**Comment by vrozov** [comment]
Please fix license header.

**Comment by pramodin** [comment]
@vrozov fixed the header

**Comment by pramodin** [comment]
@vrozov anything else

File: engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
   public static void init(Configuration configuration)
   {
-    init(configuration, null);
+    init(configuration, StramHTTPAuthentication.FOLLOW_HADOOP_AUTH);
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java]: Why is this change necessary? Does `null` corresponds to default and is the same as  `StramHTTPAuthentication.FOLLOW_HADOOP_AUTH`?

Is static block on line 44 necessary? Can `stramWebSecurityEnabled` be explicitly initialized to `UserGroupInformation.isSecurityEnabled()`?

File: engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java [hunk 2]
```diff
@@ -59,16 +59,14 @@ public static void init(Configuration configuration, StramHTTPAuthentication str
...
-      if (stramHTTPAuth == Context.StramHTTPAuthentication.FOLLOW_HADOOP_HTTP_AUTH) {
-        stramWebSecurityEnabled = hadoopWebSecurityEnabled;
-      } else if (stramHTTPAuth == StramHTTPAuthentication.FOLLOW_HADOOP_AUTH) {
-        stramWebSecurityEnabled = UserGroupInformation.isSecurityEnabled();
-      } else if (stramHTTPAuth == StramHTTPAuthentication.ENABLE) {
-        stramWebSecurityEnabled = true;
-      } else if (stramHTTPAuth == StramHTTPAuthentication.DISABLE) {
-        stramWebSecurityEnabled = false;
-      }
+    if (stramHTTPAuth == Context.StramHTTPAuthentication.FOLLOW_HADOOP_HTTP_AUTH) {
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java]: Please change to `StramHTTPAuthentication.FOLLOW_HADOOP_HTTP_AUTH` to be consistent with the remaining if statements

File: engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java [hunk 3]
```diff
@@ -59,16 +59,14 @@ public static void init(Configuration configuration, StramHTTPAuthentication str
...
+      stramWebSecurityEnabled = hadoopWebSecurityEnabled;
+    } else if (stramHTTPAuth == StramHTTPAuthentication.FOLLOW_HADOOP_AUTH) {
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java]: Should this case be handled as a catch all?

File: engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java (hunk 1 above)
Comment (pramodin) [reply engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java]: null corresponds to rely on defaults (set by static) and not set stramWebSecurityEnabled. However, this causes issue with multiple tests where init can be called multiple times even though in reality this won't happen.

File: engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java (hunk 3 above)
Comment (pramodin) [reply engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java]: The statics should handle the null case

File: engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java (hunk 2 above)
Comment (pramodin) [reply engine/src/main/java/com/datatorrent/stram/util/SecurityUtils.java]: Done

**Comment by pramodin** [comment]
@vrozov addressed comments
